components/net/filemanager_thread.rs

```diff
@@ -90,14 +95,14 @@ impl FileManager {
         origin: FileOrigin,
     ) {
         let store = self.store.clone();
-        thread::Builder::new()
-            .name("read file".to_owned())
-            .spawn(move || {
+        self.thread_pool.upgrade().and_then(|pool| {
```

Comment:
what if the upgrade fails?

Reply:
It would mean the system has exited and a worker was still running after that, and then tried to spawn additional work(fetch for example uses the pool from work running on the pool). So I was thinking of not handling it since there seems to be no way to recover from it.

Reply:
Right, but we should have a `warn!` or something

Reply:
Good idea.